test/test-utils/phase-interceptor.ts

```diff
@@ -1,450 +1,224 @@
-  name: string;
-  callback(): void;
-  onError(error: any): void;
-}
-
-interface onHoldStub {
-  name: string;
-  call(): void;
-}
-
+// biome-ignore-start lint/correctness/noUnusedImports: TSDoc imports
+import type { GameManager } from "#test/test-utils/game-manager";
+import type { PromptHandler } from "#test/test-utils/helpers/prompt-handler";
+import type { Constructor } from "#types/common";
+// biome-ignore-end lint/correctness/noUnusedImports: TSDoc imports
+import { inspect } from "util";
+import chalk from "chalk";
+import { vi } from "vitest";
+import { getEnumStr } from "./string-utils";
+
+/**
+ * The interceptor's current state.
+ * Possible values are the following:
+ * - `running`: The interceptor is currently running a phase.
+ * - `interrupted`: The interceptor has been interrupted by a UI prompt or similar mechanism,
+ *    and is currently waiting for the current phase to end.
+ * - `idling`: The interceptor is not currently running a phase and is ready to start a new one.
+ */
+type StateType = "running" | "interrupted" | "idling";
+
+/**
+ * The PhaseInterceptor is a wrapper around the `BattleScene`'s {@linkcode PhaseManager}.
+ * It allows tests to exert finer control over the phase system, providing logging, manual advancing, and other helpful utilities.
+ */
 export class PhaseInterceptor {
-  public scene: BattleScene;
-  // @ts-expect-error: initialized in `initPhases`
-  public phases: Record<PhaseString, PhaseStub> = {};
-  public log: PhaseString[];
+  private scene: BattleScene;
   /**
-   * TODO: This should not be an array;
-   * Our linear phase system means only 1 phase is ever started at once (if any)
+   * A log containing all phases having been executed in FIFO order. \
+   * Entries are appended each time {@linkcode run} is called, and can be cleared manually with {@linkcode clearLogs}.
    */
-  private onHold: onHoldStub[];
-  private interval: NodeJS.Timeout;
-  private promptInterval: NodeJS.Timeout;
-  private intervalRun: NodeJS.Timeout;
-  private prompts: PromptHandler[];
-  private inProgress?: InProgressStub;
-  private originalSetMode: typeof UI.prototype.setMode;
-  private originalSuperEnd: typeof Phase.prototype.end;
-
+  public log: PhaseString[] = [];
   /**
-   * List of phases with their corresponding start methods.
-   *
-   * CAUTION: If a phase and its subclasses (if any) both appear in this list,
-   * make sure that this list contains said phase AFTER all of its subclasses.
-   * This way, the phase's `prototype.start` is properly preserved during
-   * `initPhases()` so that its subclasses can use `super.start()` properly.
+   * The interceptor's current state.
+   * Possible values are the following:
+   * - `running`: The interceptor is currently running a phase.
+   * - `interrupted`: The interceptor has been interrupted by a UI prompt and is waiting for the caller to end it.
+   * - `idling`: The interceptor is not currently running a phase and is ready to start a new one.
```

Comment:
```suggestion
   * @see {@linkcode StateType}
```
Duplicated documentation tends to get out of sync very quickly.